synapse/storage/engines/sqlite.py

```diff
@@ -30,6 +31,11 @@ def __init__(self, database_module, database_config):
         database = database_config.get("args", {}).get("database")
         self._is_in_memory = database in (None, ":memory:",)
 
+        if platform.python_implementation() == "PyPy":
+            # related to #6186, pypy's sqlite3 module doesn't like bytearrays,
+            # so here we're converting them back
+            database_module.register_adapter(bytearray, lambda array: bytes(array))
```

Comment:
Back when i made this change, there were still a few of them left, and so I made this change to assure they were caught anyway.

I can remove this, but i'd rather not, because i think it is pretty possible `bytearray`s are re-introduced somewhere else, which then gets caught on this.

Reply:
Fair enough.